orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/ManifestForceCacheRefreshTask.java

```diff
@@ -112,6 +112,9 @@ public TaskResult execute(@Nonnull Stage stage) {
     StageData stageData = fromStage(stage);
     stageData.deployedManifests = getDeployedManifests(stage);
 
+    // Clear manifests to be refreshed
+    stageData.manifestNamesByNamespaceToRefresh = null;
```

Comment:
Okay I'm a dummy, this is breaking the tests because members of an immutable map can't be null, and the outputs of a task are an immutable map:
https://github.com/spinnaker/orca/blob/master/orca-core/src/main/java/com/netflix/spinnaker/orca/TaskResult.java#L51
Going to sleep on this one, we'll need a different way to somehow mark this manifest as refreshed. Otherwise, when we have a subsequent disable stage as part of red/black, we'll be trying to refresh any manifests removed as part of the cleanup rather than the manifest to disable (because enable/disable force cache refresh tasks are not preceded by the promote outputs task that adds the deployed manifests to the refresh key).

Reply:
@ezimanyi since we cannot null out context keys, I added a flag, `shouldRefreshManifestNamesByNamespaceToRefresh`, which is set to true in the `PromoteManifestKatoOutputsTask` when a task produces manifest outputs, and then set to false in the `ManifestForceCacheRefreshTask` when those manifests are refreshed in the cache. While kind of disgusting, this should keep this change fully backwards compatible.